data/pigui/libs/equip-card.lua

```diff
@@ -0,0 +1,208 @@
+--
+-- DrawEquipmentItem* functions take a "data" argument table with the form:
+--   icon  - icon index to display for the slot
+--   name  - translated name to display on the slot
+--   equip - equipment object being drawn (or none for an empty slot)
+--   type* - translated "slot type" name to display
+--   size* - string of the form "S1" to display in the slot name for sized slots
+--   stats - list of detailed item stats to display in the item tooltip,
+--           in the format { label, icon, value, color }
+--   [...] - up to 4 { icon, value, tooltip } data items for the stats line
+
+---@class UI.EquipCard : UI.ItemCard
+---@field New fun(): self
+local EquipCard = utils.inherits(ItemCard, "UI.EquipCard")
+
+---@class UI.EquipCard.Data
+---@field icon ui.Icon
+---@field name string?
+---@field count integer?
+---@field type string?
+---@field size string?
+---@field equip EquipType?
+---@field slot HullConfig.Slot?
+---@field present integer?
+---@field total integer?
+---@field stats EquipType.UI.Stats[] | nil
+
+EquipCard.tooltipStats = true
+EquipCard.highlightBar = true
+EquipCard.detailFields = 4
+EquipCard.tooltipWrapEm = 18
+
+EquipCard.emptyIcon = icons.autopilot_dock
+
+local tooltipStyle = {
+	WindowPadding = ui.theme.styles.WindowPadding,
+	WindowRounding = EquipCard.rounding
+}
+
+---@param data UI.EquipCard.Data
+function EquipCard:tooltipContents(data, isSelected)
+	if not data.equip then
+		return
+	end
+
+	ui.withFont(pionillium.body, function()
+		ui.text(data.equip:GetName())
+
+		local desc = data.equip:GetDescription()
+
+		if desc and desc ~= "" then
+			ui.withFont(pionillium.details, function()
+				ui.spacing()
+				ui.textWrapped(desc)
+			end)
+		end
+	end)
+
+	if self.tooltipStats and data.stats then
+		ui.spacing()
+		ui.separator()
+		ui.spacing()
+
+		ui.withFont(pionillium.details, function()
+			self.drawEquipStats(data)
+		end)
+	end
+end
+
+---@param data UI.EquipCard.Data
+function EquipCard:drawTooltip(data, isSelected)
+	if not (self.tooltipStats and data.stats or data.equip) then
+		return
+	end
+
+	ui.withStyleVars(tooltipStyle, function()
+		ui.customTooltip(function()
+			local wrapWidth = ui.getTextLineHeight() * self.tooltipWrapEm
+
+			ui.pushTextWrapPos(wrapWidth)
+
+			self:tooltipContents(data, isSelected)
+
+			ui.popTextWrapPos()
+		end)
+	end)
+end
+
+---@param data UI.EquipCard.Data
+function EquipCard.drawEquipStats(data)
+	if ui.beginTable("EquipStats", 2, { "NoSavedSettings" }) then
+
+		ui.tableSetupColumn("##name", { "WidthStretch" })
+		ui.tableSetupColumn("##value", { "WidthFixed" })
+
+		ui.pushTextWrapPos(-1)
+		for i, tooltip in ipairs(data.stats) do
+			ui.tableNextRow()
+
+			ui.tableSetColumnIndex(0)
+			ui.text(tooltip[1] .. ":")
+
+			ui.tableSetColumnIndex(1)
+			ui.icon(tooltip[2], Vector2(ui.getTextLineHeight(), ui.getTextLineHeight()), ui.theme.colors.font)
+			ui.sameLine(0, ui.getTextLineHeight() / 4.0)
```

Comment:
What is the magic-number `4.0` for?

Reply:
Just a spacing constant. Could/should probably be `styles.ItemInnerSpacing`.

Reply:
Sorry to be nitpicky; would just be nice for new code to adhere to Best Practices :)